base/docs/basedocs.jl

```diff
@@ -2281,6 +2282,21 @@ instruction, otherwise it'll use a loop.
 """
 replacefield!
 
+"""
+    setfieldonce!(value, name::Union{Int,Symbol}, desired,
+                  [success_order::Symbol, [fail_order::Symbol=success_order]) -> success::Bool
+
+Atomically perform the operations to set a field to
+a given value, only if it was previously not set.
+
+    ok = !isdefined(value, name, fail_order)
+    if ok
+        setfield!(value, name, desired, success_order)
+    end
+    return ok
+"""
```

Comment:
Probably should also get a compat note?

Reply:
Hopefully I got all those versions right